examples/ShowPlayer.cpp

```diff
@@ -85,68 +92,111 @@ int ShowPlayer::Playback(unsigned int iterations,
+  ShowEntry entry;
+  unsigned int playhead_time = m_playback_pos;
+  do {
+    ShowLoader::State state = m_loader.NextEntry(&entry);
+    switch (state) {
+      case ShowLoader::END_OF_FILE:
+        HandleEndOfFile();
+        return;
+      case ShowLoader::INVALID_LINE:
+        m_client.GetSelectServer()->Terminate();
+        return;
+      default: {}
+    }
+    playhead_time += entry.next_wait;
+  } while (playhead_time < seek_time);
+  // Adjust the timeout to handle landing in the middle of the entry's timeout
+  m_playback_pos = seek_time;
+  entry.next_wait = playhead_time - seek_time;
+  SendEntry(entry);
 }
 
 
 /**
- * Get the next time offset
+ * Send the next frame in the show file
  */
-ShowLoader::State ShowPlayer::RegisterNextTimeout() {
-  unsigned int timeout;
-  ShowLoader::State state = m_loader.NextTimeout(&timeout);
-  if (state != ShowLoader::OK) {
-    return state;
+void ShowPlayer::SendNextFrame() {
+  ShowEntry entry;
+  ShowLoader::State state = m_loader.NextEntry(&entry);
+  if (state == ShowLoader::END_OF_FILE ||
+      (m_stop > 0 && m_playback_pos >= m_stop)) {
+    if (entry.buffer.Size() > 0) {
+      // Send the last frame before looping/exiting
+      SendFrame(entry);
+    }
+    HandleEndOfFile();
+    return;
+  } else if (state == ShowLoader::INVALID_LINE) {
+    OLA_FATAL << "Invalid data at line " << m_loader.GetCurrentLineNumber();
+    m_client.GetSelectServer()->Terminate();
+    return;
   }
+  SendEntry(entry);
+}
 
-  OLA_INFO << "Registering timeout for " << timeout << "ms";
+
+/**
+ * Send @p entry and wait for next
+ * @param entry the show file entry to send
+ */
+void ShowPlayer::SendEntry(const ShowEntry &entry) {
+  // Send DMX data
+  SendFrame(entry);
+
+  // Set when next to send data
+  OLA_INFO << "Registering timeout for " << entry.next_wait << "ms";
+  m_playback_pos += entry.next_wait;
   m_client.GetSelectServer()->RegisterSingleTimeout(
-      timeout,
-      ola::NewSingleCallback(this, &ShowPlayer::SendNextFrame));
-  return state;
+     entry.next_wait,
+     ola::NewSingleCallback(this, &ShowPlayer::SendNextFrame));
+}
+
+
+void ShowPlayer::SendFrame(const ShowEntry &entry) const {
+  OLA_INFO << "Universe: " << entry.universe << ": " << entry.buffer.ToString();
+  ola::client::SendDMXArgs args;
+  m_client.GetClient()->SendDMX(entry.universe, entry.buffer, args);
 }
 
 
 /**
  * Handle the case where we reach the end of file
  */
 void ShowPlayer::HandleEndOfFile() {
+  if (m_stop > 0 && m_playback_pos < m_stop) {
+    OLA_WARN << "Show file ends before the stop time (Actual length "
+             << m_playback_pos << " ms)";
+  }
   m_iteration_remaining--;
   if (m_infinite_loop || m_iteration_remaining > 0) {
     m_loader.Reset();
-    m_client.GetSelectServer()->RegisterSingleTimeout(
-        m_loop_delay,
-        ola::NewSingleCallback(this, &ShowPlayer::SendNextFrame));
+    // Move to start point and send the frame
+    SeekTo(m_start);
```

Comment:
We should still honour any loop delay that's been set, why don't you just call the new Start function?